### PR TITLE
feat: delete orphan charts when updating dashboard

### PR DESCRIPTION
**Pull request by ZeRego**
Closes: #6394 

### Description:

- [x] after dashboard update, delete all unused charts that belong to that dashboard
- [x] track 'saved_chart.deleted` event per each one

https://github.com/lightdash/lightdash/assets/9117144/49895ef5-292b-4b59-a5af-769ec4a1d46f

### CODE REVIEW

File: packages/backend/src/models/DashboardModel/DashboardModel.ts [hunk 1]
```diff
@@ -773,4 +774,34 @@ export class DashboardModel {
...
+                DashboardVersionsTableName,
+                `${DashboardsTableName}.dashboard_id`,
+                `${DashboardVersionsTableName}.dashboard_id`,
+            )
+            .select([`${DashboardVersionsTableName}.dashboard_version_id`])
+            .where(`${DashboardsTableName}.dashboard_uuid`, dashboardUuid)
+            .orderBy(`${DashboardVersionsTableName}.created_at`, 'desc')
+            .limit(1);
+
+        const getChartsInTilesQuery = this.database(DashboardTileChartTableName)
```
Comment (rephus) [inline packages/backend/src/models/DashboardModel/DashboardModel.ts:792]: missing await ?

File: packages/backend/src/models/DashboardModel/DashboardModel.ts [hunk 2]
```diff
@@ -773,4 +774,34 @@ export class DashboardModel {
         });
         return this.getById(dashboardUuid);
     }
+
+    async getOrphanedCharts(
+        dashboardUuid: string,
+    ): Promise<Pick<SavedChart, 'uuid'>[]> {
+        const getLastVersionIdQuery = this.database(DashboardsTableName)
```
Comment (rephus) [inline packages/backend/src/models/DashboardModel/DashboardModel.ts:781]: missing await ?

File: packages/backend/src/services/DashboardService/DashboardService.ts [hunk 3]
```diff
@@ -87,6 +93,30 @@ export class DashboardService {
...
+        );
+        await Promise.all(
+            orphanedCharts.map(async (chart) => {
+                const deletedChart = await this.savedChartModel.delete(
+                    chart.uuid,
+                );
+                analytics.track({
+                    event: 'saved_chart.deleted',
+                    userId: user.userUuid,
+                    properties: {
```
Comment (rephus) [inline packages/backend/src/services/DashboardService/DashboardService.ts:111]: should we add an special flag for this saved_chart.deleted ?

File: packages/backend/src/models/DashboardModel/DashboardModel.ts (hunk 1 above)
Comment (ZeRego) [reply packages/backend/src/models/DashboardModel/DashboardModel.ts:792]: No, I just want to prepare/generate these subqueries separately for better readability. 
I only await the last query in this function which includes all these subqueries.

File: packages/backend/src/services/DashboardService/DashboardService.ts (hunk 3 above)
Comment (ZeRego) [reply packages/backend/src/services/DashboardService/DashboardService.ts:111]: Going to introduce that in the next ticket https://github.com/lightdash/lightdash/issues/6388

**Review by rephus — APPROVED** [review]

**Pull request merged**